Initial support for building on Ampere GPU, CUDA 11, cuDNN 8

zasdfgbnm opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #39361 [WIP] Enable TF32 support for cuDNN
* #39283 [WIP] Enable TF32 support for cuBLAS
* **#39277 Initial support for building on Ampere GPU, CUDA 11, cuDNN 8**

This PR contains initial changes that makes PyTorch build with Ampere GPU, CUDA 11, and cuDNN 8.
TF32 related features will not be included in this PR.

Differential Revision: [D21832814](https://our.internmc.facebook.com/intern/diff/D21832814)

zasdfgbnm committed:
Initial support for building on Ampere GPU, CUDA 11, cuDNN 8

This PR contains initial changes that makes PyTorch build with Ampere GPU, CUDA 11, and cuDNN 8.
TF32 related features will not be included in this PR.

[ghstack-poisoned]

zasdfgbnm commented:
cc: @ptrblck

zasdfgbnm committed:
Update on "Initial support for building on Ampere GPU, CUDA 11, cuDNN 8"


This PR contains initial changes that makes PyTorch build with Ampere GPU, CUDA 11, and cuDNN 8.
TF32 related features will not be included in this PR.

[ghstack-poisoned]

ezyang commented:
Can you rebase on master? I want this built on a clean ROCm version just to see if I have to phone up AMD.

zasdfgbnm committed:
Update on "Initial support for building on Ampere GPU, CUDA 11, cuDNN 8"


This PR contains initial changes that makes PyTorch build with Ampere GPU, CUDA 11, and cuDNN 8.
TF32 related features will not be included in this PR.

[ghstack-poisoned]

zasdfgbnm commented:
@ezyang rebased

zasdfgbnm commented:
@ezyang @malfet The ROCm CIs are passing, and the two failures are unrelated (unable to find GPG key). This should be working.

malfet commented:
Triggered re-run on failed builds

malfet commented on torch/utils/cpp_extension.py:
```diff
@@ -1375,7 +1375,7 @@ def _get_cuda_arch_flags(cflags=None):
     ])
```
```suggestion
        ('Ampere', '8.0+PTX'),
    ])
```

zasdfgbnm replied:
Change applied. Thanks for pointing out!

zasdfgbnm committed:
Update on "Initial support for building on Ampere GPU, CUDA 11, cuDNN 8"


This PR contains initial changes that makes PyTorch build with Ampere GPU, CUDA 11, and cuDNN 8.
TF32 related features will not be included in this PR.

[ghstack-poisoned]

zasdfgbnm commented:
All CI are passing
